appdynamics/startAppDynamics.sh

```diff
@@ -0,0 +1,438 @@
+# using the agent installation.
+# 
+# If this property is specified, all agent logs are written to <Agent-Runtime-Directory>/logs/node-name 
+# and transaction configuration is written to the <Agent-Runtime-Directory>/conf/node-name directory.
+#
+# Where not supplied, defaults to ${APPDYNAMICS_AGENT_HOME}/defaultAgentVersion.
+#
+# Example: APPDYNAMICS_AGENT_BASE_DIR="${APPDYNAMICS_AGENT_HOME}/22.12.0.34603"
+#
+#
+# APPDYNAMICS_PROXY_OPTS: Specify system properties for proxy host/port if using a proxy to connect to the 
+# AppDynamics Controller.
+#
+# Example:  APPDYNAMICS_PROXY_OPTS="-Dappdynamics.http.proxyHost=<proxy host> -Dappdynamics.http.proxyPort=<proxy port>"
+#
+# No proxy required by default.
+#
+
+# The location of the Java AppDynamics Java Agent files
+APP_AGENT_REPO_HOME="/opt/appdynamics/AppServerAgent"
+
+source ${APP_AGENT_REPO_HOME}/logging_functions  
+
+
+# =============================================================================
+# functions: 
+# =============================================================================
+
+# 
+# Checks for existence of any PIDs representing running Weblogic Server JVM instances.
+# Exits if encounters invalid state of more than one PID found.
+# Returns 0 if no Weblogic Server JVM instance PID found; or
+# Returns the PID of any single Weblogic Server JVM instance PID found
+#
+f_getWeblogicServerPID() {
+  # Look for Weblogic Server  
+  javaWeblogicServer=$(pgrep --full --delimiter ',' "weblogic.Server")
+
+  # String not empty and contains comma => illegal state 
+  if [[ -n "${javaWeblogicServer}" ]] && [[ "${javaWeblogicServer}" == *","* ]]
+  then 
+    exit 1 # Error, illegal state
+  elif [[ -n "${javaWeblogicServer}" ]] # Single Server PID detected  
+  then 
+    echo "$javaWeblogicServer"
+  else # No server PID detected  
+    echo 0    
+  fi  
+}
+
+#
+# Checks the PID supplied is still the  current Weblogic Server instance
+#
+# Arg 1: Prior PID
+#
+# Returns 0 if true, 1 if false or no PIDS running
+#
+f_isSameWeblogicServerPID() {
+  typeset priorPID=$1; shift
+  currentPID=$(f_getWeblogicServerPID)
+
+  if [[ -n "${currentPID}" ]]  &&  [[ "${currentPID}" == "${priorPID}" ]]; then
+    echo 0 #true    
+  else    
+    echo 1 #false
+  fi
+  
+}
+
+
+#
+# Checks the conditions to enable AppDynamics monitoring
+# Returns true if allowed, else false.
+#
+f_initAppDynamicsEnabled() {
+  if [[ -z "${APPDYNAMICS_AGENT_ENABLED}" ]]
+  then 
+    f_logWarn "APPDYNAMICS_AGENT_ENABLED not set. Will be disabled by default."
+    APPDYNAMICS_AGENT_ENABLED=false
+  fi
+
+  if [[ "${APPDYNAMICS_AGENT_ENABLED}" == true ]]
+  then
+    # Decide if env supports use based on match to naming of a higher environment
+    if [[ -n "${ENVIRONMENT_LABEL}" ]]
+    then 
+
+      # Check var contains either Live or Staging
+      if echo "$ENVIRONMENT_LABEL" | grep -q "Live\|Staging";
+      then
+        APPDYNAMICS_AGENT_ENABLED=true
```

Comment:
This line is probably redundant, as APPDYNAMICS_AGENT_ENABLED=true already at this point.

Reply:
Will fix as need to cater further for where no appd zip available for local build

Reply:
Nice work!